src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -158,6 +181,795 @@ MultiLogicalPlanCreate(Query *originalQuery, Query *queryTree)
 
 
 /*
+ * MultiSubqueryPlanTree gets the query objects and returns logical plan
+ * for subqueries.
+ *
+ * We currently have two different code paths for creating logic plan for subqueries:
+ *   (i) subquery pushdown
+ *   (ii) single relation repartition subquery
+ *
+ * In order to do create the logical plan, we follow the algorithm below:
+ *    -  If subquery pushdown planner can plan the query
+ *        -  We're done, we create the multi plan tree and return
+ *    -  Else
+ *       - If the query is not eligible for single table repartition subquery planning
+ *            - Throw the error that the subquery pushdown planner generated
+ *       - If it is eligible for single table repartition subquery planning
+ *            - Check for the errors for single table repartition subquery planning
+ *                - If no errors found, we're done. Create the multi plan and return
+ *                - If found errors, throw it
```

Comment:
> I can think of union/union all queries on a single table. However, If you're comfortable with this approach, I'm okay to continue with this given that you know the example queries better than me.

Each occurrence of a relation generates a separate `RTE_RELATION`, and also since single-repartition queries do not support `UNIONs`, we'd error out using pushdown errors.